src/builder-context.h

```diff
@@ -22,14 +22,16 @@
 #define __BUILDER_CONTEXT_H__
 
 #include <gio/gio.h>
-#include <libsoup/soup.h>
 #include <curl/curl.h>
 #include "builder-options.h"
 #include "builder-utils.h"
 #include "builder-sdk-config.h"
 
 G_BEGIN_DECLS
 
+/* Same as SOUP_HTTP_URI_FLAGS, means all possible flags for http uris */
+#define CONTEXT_HTTP_URI_FLAGS (G_URI_FLAGS_HAS_PASSWORD | G_URI_FLAGS_ENCODED_PATH | G_URI_FLAGS_ENCODED_QUERY | G_URI_FLAGS_ENCODED_FRAGMENT | G_URI_FLAGS_SCHEME_NORMALIZE)
```

Comment:
`GUri` with `G_URI_FLAGS_SCHEME_NORMALIZE` requires bumping `GLIB_REQS` in `configure.ac` to 2.68 (see flatpak/flatpak#5062 and flatpak/flatpak#5064), which would rule out compiling upstream source on Debian 11.

Alternatively, if it's OK to use `G_URI_FLAGS_SCHEME_NORMALIZE` if and only if it exists (as in flatpak/flatpak#5064), then that use of `GUri` will depend on GLib 2.66, which is available in Debian 11.

Reply:
Yeah, I know the failure reason. Trying to raise some discussion on it. It is clear flatpak needs to support ancient distros but with flatpak-builder situation has been more fuzzy.

Reply:
But yeah, maybe not raising bar above 2.66 would be a good call anyway. This build runs with Ubuntu 20.04 which I suppose has even older glib.

Reply:
Right. I now did this change, there's obvious error for logs `Requested 'glib-2.0 >= 2.66' but version of GLib is 2.64.6`

Reply:
Does flatpak require glib 2.66 to build?

Reply:
Flatpak itself has a backport of GUri for GLib older than 2.66. The problem in flatpak/flatpak#5062 is that if GLib is *exactly* 2.66, as in Debian 11, the backport isn't used but the real OS GLib doesn't support `G_URI_FLAGS_SCHEME_NORMALIZE`.

Reply:
> It is clear flatpak needs to support ancient distros but with flatpak-builder situation has been more fuzzy

Flatpak allegedly supports ancient distros, but it doesn't *actually* support (sufficiently) ancient distros: flatpak/flatpak#5066.

For flatpak-builder and other ancillary stuff, I personally think a reasonable place to draw the line is the latest Debian stable release or the latest Ubuntu LTS, whichever is older. They're both released on a cycle of about 2 years, with an offset of 1 year, so the one that is the limiting factor swaps periodically: it's currently Debian 11, but when Debian 12 is released, Ubuntu 22.04 will become the limiting factor.

For Flatpak itself, clearly there's interest in supporting older distros in order to be able to use Flatpak to run recent software on old distros, and in particular people do use it on older-than-latest-LTS branches of Ubuntu (although I'm the only one who is actually working on those backports, and I don't use Ubuntu myself). I think we should seriously consider dropping any attempt to backport to Ubuntu 16.04 (which is EOL for Ubuntu's standard support, and requires a support subscription for extended security maintenance), and perhaps also 18.04 (which is not EOL, but Flatpak doesn't compile successfully there and nobody noticed until I tried to update the backport).

Reply:
Seems there is no flatpak backports for jammy.

Reply:
I did not touch that part of build that did `sudo add-apt-repository ppa:alexlarsson/flatpak` yet.

Reply:
Appears build is also doing `sudo add-apt-repository 'deb https://download.mono-project.com/repo/ubuntu stable-jammy main' # Needed for updates to work`. I updated name, seems there is no package for jammy. I don't know what we need from there.

Reply:
Adding the Mono repository was a workaround for a bug in how Github set up their containers: see <https://github.com/flatpak/flatpak/pull/4299>. If we don't need to use that workaround any more in 22.04, then it's better to stop using it (remove that line). We only want that workaround if it's still necessary.

Reply:
It's probably actually better if we do each CI build with a minimum of backports, and only add extra non-Ubuntu repositories as we find that we need them. Ubuntu 22.04 has Flatpak 1.12.x, which seems like it should be sufficient for flatpak-builder for a while.

Reply:
Looks like Python 2 test is skipped completely currently since build machine does not have Python 2 installed.

Reply:
@smcv I thought I did the requested changes but GitHub is still complaining. Did I miss something?